ChaosMod/Components/EffectDispatcher.cpp

```diff
@@ -125,15 +127,11 @@ void EffectDispatcher::UpdateEffects()
 			}
 
 			if ((effect.m_fMaxTime > 0 && effect.m_fTimer <= 0)
-				|| ((!effectData.IsMeta)
-					&& (effect.m_fTimer < -m_usEffectTimedDur + (activeEffectsSize > 3
-						? ((activeEffectsSize - 3) * 20 < 160
-							? (activeEffectsSize - 3) * 20
-							: 160)
-						: 0))))
+				|| (!effectData.IsMeta && (activeEffectsSize > maxEffects  || (effect.m_fMaxTime > 0 && ShouldRemoveEffectForTimeOut(effect.m_fTimer, activeEffectsSize, effectCountToCheckCleaning))))
+				)
```

Comment:
It's probably worth splitting this into a few variables, since this this statement is a bit difficult to parse

Reply:
But it's definitely better than the nested ternaries

Reply:
done